src/App/Page.php

```diff
@@ -30,15 +30,26 @@ class Page implements ArrayAccess
 	/**
 	 * @var array Contains all stylesheets, which should get loaded during page
 	 */
-	private $stylesheets;
+	private $stylesheets = [];
 	/**
 	 * @var array Contains all scripts, which are added to the footer at last
 	 */
-	private $footerScripts;
+	private $footerScripts = [];
 	/**
 	 * @var array The page content, which are showed directly
 	 */
-	private $page;
+	private $page = [
```

Comment:
Why do we need to automatically initialize the fields instead of the constructor? The constructor should be the place where initializing should stay. Same goes for the script arrays.

Reply:
I don't think so. In the constructor we should assign constructor parameters, but PHP gives the opportunity to do the initialization when the members are declared, which avoids forgetting to initialize the members in the constructor like it was the case for half the members of this class.

So in this convenient spot you have:
- member declaration
- type-hinting
- default value

And in the constructor you just need to set the constructor parameters that are explicitly there, unlike the class members that are declared elsewhere.

Reply:
Hmm sounds reasonable. So we should take a the other classes as well to be consistent at declarating default values. Maybe I can have a look.